M2/Macaulay2/m2/modules.m2

```diff
@@ -151,6 +155,10 @@ new Module from Sequence := (Module,x) -> (
      	       symbol numgens => rawRank rM
      	       })) x
 
+vector(Module, List) := (M, v) -> vector map(M,,apply(splice v, i -> {i}))
+vector(Module, RingElement) :=
+vector(Module, Number)      := (M, x) -> vector(M, {x})
```

Comment:
It might also be worth adding `vector(Module, Matrix)` and changing how vectors are printed with `peek` to use that instead of this:
```m2
i36 : v

o36 = | 1 |
      | 1 |

o36 : cokernel | 2 0 |
               | 0 2 |

i37 : peek v

o37 = cokernel | 2 0 |{| 1 |}
               | 0 2 | | 1 |
```

Reply:
So `vector(Module,  Matrix)` would be something like `(M, f) -> vector map(M, ,f)`?

What are you suggesting for `peek`?  `peek` (which wraps around `peek'`) is pretty low-level and is currently interpreting each `Vector` object as its parent `BasicList`.  There don't appear to be any other mathematical objects with their own `peek'` methods.

Reply:
Ah, true about `peek`. Then let's fix `toString` and friends at least.

> So `vector(Module, Matrix)` would be something like `(M, f) -> vector map(M, ,f)`?

Yes, and perhaps `vector(Module, List)` can call this for simplicity.

Reply:
`toString(Vector)` is updated now:

```m2
i1 : toString vector {1, 2, 3}

o1 = vector(ZZ^3, {1, 2, 3})
```

I've also added `vector(Module, Matrix)`.  I kept `vector(Module, List)` alone since `map(Module,Nothing,List)` does a bunch of the heavy lifting for us.

Reply:
Hmm, the difference between `map(Module, Nothing, List)` and `map(Module, Nothing, Matrix)` is kind of mysterious to me. Does this also fix https://github.com/Macaulay2/M2/issues/3016#issuecomment-1830305241?

Reply:
Yeah, I haven't looked at the code very much.  I just noticed that one did the thing I wanted and the other didn't:

```m2
i1 : R = QQ[x];

i2 : vector map(R^1,, matrix {{5}})
stdio:2:8:(3): error: expected module and matrix for the same ring

i3 : vector map(R^1,, {{5}})

o3 = | 5 |

      1
o3 : R
```